pkg/security/certificateSigner/certificateSigner.go

```diff
@@ -0,0 +1,118 @@
+package certificateSigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/plgd-dev/kit/v2/security"
+)
+
+type SignerConfig struct {
+	ValidNotBefore       time.Time
+	ValidNotAfter        time.Time
+	OverrideCertTemplate func(template *x509.Certificate) error
+}
+
+type Opt = func(cfg *SignerConfig)
+
+func WithNotBefore(validNotBefore time.Time) Opt {
+	return func(cfg *SignerConfig) {
+		cfg.ValidNotBefore = validNotBefore
+	}
+}
+
+func WithNotAfter(validNotAfter time.Time) Opt {
+	return func(cfg *SignerConfig) {
+		cfg.ValidNotAfter = validNotAfter
+	}
+}
+
+func WithOverrideCertTemplate(overrideCertTemplate func(template *x509.Certificate) error) Opt {
+	return func(cfg *SignerConfig) {
+		cfg.OverrideCertTemplate = overrideCertTemplate
+	}
+}
+
+type CertificateSigner struct {
+	caCert []*x509.Certificate
+	caKey  crypto.PrivateKey
+	cfg    SignerConfig
+}
+
+func New(caCert []*x509.Certificate, caKey crypto.PrivateKey, opts ...Opt) *CertificateSigner {
+	cfg := SignerConfig{
+		ValidNotAfter: time.Unix(1<<63-62135596801, 999999999),
```

Comment:
Just out of curiousity - what time is this?

Reply:
Is MaxTime - golang doesn't have constant 😭

Reply:
292277026596-12-04 15:30:07.999999999 +0000 UTC

Reply:
I learn something everyday. :D